### PR TITLE
Add main packs and centralize dependencies versions

### PR DESCRIPTION
**Pull request by tdelhomenie-microej**

### CODE REVIEW

File: vee-port/extensions/front-panel/build.gradle.kts [hunk 1]
```diff
@@ -7,6 +7,7 @@ microej {
 }
 
 dependencies {
+    implementation(libs.frontpanel.widget)
```
Comment (gbalan-microej) [inline vee-port/extensions/front-panel/build.gradle.kts:10]: is it possible to remove the lines 
```
   implementation(libs.pack.ui) {
        artifact {
            name = "frontpanel"
            extension = "jar"
        }
```
it denotes an advanced use case which is not required here according to me

File: vee-port/configuration.properties [hunk 2]
```diff
@@ -122,4 +123,4 @@
 # Expected value is one among this list: ARGB8888 | RGB888 | RGB565 | ARGB1555 | ARGB4444 | 16 | 24 | 32.
```
Comment (gbalan-microej) [inline vee-port/configuration.properties:123]: the latest comment in the UI pack is:
```
# Defines the number of bits per pixels the display device is using to render a pixel.
# Expected value is one among these both list: 
# Standard formats:
# - ARGB8888: Alpha 8 bits; Red 8 bits; Green 8 bits; Blue 8 bits 
# - RGB888: Alpha 0 bit; Red 8 bits; Green 8 bits; Blue 8 bits (fully opaque)
# - RGB565: Alpha 0 bit; Red 5 bits; Green 6 bits; Blue 5 bits (fully opaque)
# - ARGB1555: Alpha 1 bit; Red 5 bits; Green 5 bits; Blue 5 bits (fully opaque or fully transparent)
# - ARGB4444: Alpha 4 bits; Red 4 bits; Green 4 bits; Blue 4 bits
# - C4: 4 bits to encode linear grayscale colors between 0xff000000 and 0xffffffff (fully opaque)
# - C2: 2 bits to encode linear grayscale colors between 0xff000000 and 0xffffffff (fully opaque)
# - C1: 1 bit to encode grayscale colors 0xff000000 and 0xffffffff (fully opaque)
# Custom formats:
# - 32: until 32 bits to encode Alpha, Red, Green and/or Blue
# - 24: until 24 bits to encode Alpha, Red, Green and/or Blue
# - 16: until 16 bits to encode Alpha, Red, Green and/or Blue
# - 8: until 8 bits to encode Alpha, Red, Green and/or Blue
# - 4: until 4 bits to encode Alpha, Red, Green and/or Blue
# - 2: until 2 bits to encode Alpha, Red, Green and/or Blue
# - 1: 1 bit to encode Alpha, Red, Green or Blue
# All others values are forbidden (throw a generation error).
```

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -17,9 +17,11 @@ watchdog-timer = { group = "ej.api", name = "watchdog-timer", version = "2.0.0"
 # Architecture
 architecture = { group = "com.microej.architecture.CM4.CM4hardfp_GCC48", name = "flopi4G25", version = "8.2.0" }
 # Packs
+pack-ui-architecture = { group = "com.microej.architecture.CM4.CM4hardfp_GCC48", name = "flopi4G25-ui-pack", version = "14.0.1" }
```
Comment (gbalan-microej) [inline gradle/libs.versions.toml]: use latest available pack: 14.2.0

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -17,9 +17,11 @@ watchdog-timer = { group = "ej.api", name = "watchdog-timer", version = "2.0.0"
...
 pack-ui = { group = "com.microej.pack.ui", name = "ui-pack", version = "14.0.1" }
 # Libraries
 frontpanel-framework = { group = "ej.tool.frontpanel", name = "framework", version = "1.1.0" }
+frontpanel-widget = { group = "ej.tool.frontpanel", name = "widget", version = "4.0.1" }
```
Comment (gbalan-microej) [inline gradle/libs.versions.toml:29]: use latest available version: 4.0.2

File: app/src/main/java/com/mycompany/myapplication/Main.java [hunk 5]
```diff
@@ -1,8 +1,33 @@
...
+
+        Display display = Display.getDisplay();
+        GraphicsContext g = display.getGraphicsContext();
+        int displayWidth = display.getWidth();
+        int displayHeight = display.getHeight();
+
+        g.setColor(Colors.BLACK);
+        Painter.fillRectangle(g, 0, 0, displayWidth, displayHeight);
+
+        g.setColor(Colors.GREEN);
```
Comment (gbalan-microej) [inline app/src/main/java/com/mycompany/myapplication/Main.java:25]: on a black & white display, the green may be interpreted as black color. Use WHITE instead

File: app/src/main/java/com/mycompany/myapplication/Main.java [hunk 6]
```diff
@@ -1,8 +1,33 @@
...
+        Font font = Font.getDefaultFont();
```
Comment (gbalan-microej) [inline app/src/main/java/com/mycompany/myapplication/Main.java:26]: only works if there is default font pointed by the microui.xml file

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -17,9 +17,11 @@ watchdog-timer = { group = "ej.api", name = "watchdog-timer", version = "2.0.0"
 # Architecture
 architecture = { group = "com.microej.architecture.CM4.CM4hardfp_GCC48", name = "flopi4G25", version = "8.2.0" }
 # Packs
+pack-ui-architecture = { group = "com.microej.architecture.CM4.CM4hardfp_GCC48", name = "flopi4G25-ui-pack", version = "14.0.1" }
 pack-ui = { group = "com.microej.pack.ui", name = "ui-pack", version = "14.0.1" }
```
Comment (gbalan-microej) [inline gradle/libs.versions.toml]: idem: 14.2.0 (or may be a same property ?)

File: vee-port/extensions/front-panel/src/main/resources/platform.fp [hunk 8]
```diff
@@ -5,6 +5,7 @@
 	xsi:schemaLocation="https://developer.microej.com .widget.xsd">
 
 	<device name="veeport-template" skin="splashscreen.png">
-		<!-- alias definitions start here -->
+		<ej.fp.widget.Display x="0" y="0" width="600" height="330" refreshStrategyClass="ej.fp.widget.display.brs.PredrawRefreshStrategy" bufferPolicyClass="ej.fp.widget.display.buffer.SwapTripleBufferPolicy"/>
```
Comment (gbalan-microej) [inline vee-port/extensions/front-panel/src/main/resources/platform.fp:8]: remove ` refreshStrategyClass="ej.fp.widget.display.brs.PredrawRefreshStrategy" bufferPolicyClass="ej.fp.widget.display.buffer.SwapTripleBufferPolicy"`: that will use the default settings (predraw and swap double)

File: app/src/main/java/com/mycompany/myapplication/Main.java [hunk 9]
```diff
@@ -1,8 +1,33 @@
...
+        int x = (displayWidth - font.stringWidth(HELLO_WORLD)) / 2;
+        int y = (displayHeight - font.getHeight()) / 2;
+        Painter.drawString(g, HELLO_WORLD, font, x, y);
+
+        display.flush();
```
Comment (gbalan-microej) [inline app/src/main/java/com/mycompany/myapplication/Main.java:31]: You can add a thread.sleep(5s) and then `MicroUI.stop()` to automatically close the simulator after 5s

Comment (tdelhomenie-microej) [reply app/src/main/java/com/mycompany/myapplication/Main.java:31]: Is it what we do generally in examples? We don't want the user to close the simulator manually?

File: vee-port/configuration.properties (hunk 2 above)
Comment (tdelhomenie-microej) [reply vee-port/configuration.properties:123]: OK. Should we keep `# @optional, default value is "" (empty)` ?

File: app/src/main/java/com/mycompany/myapplication/Main.java (hunk 6 above)
Comment (tdelhomenie-microej) [reply app/src/main/java/com/mycompany/myapplication/Main.java:26]: Do you suggest another solution for this example?

File: vee-port/extensions/front-panel/build.gradle.kts (hunk 1 above)
Comment (tdelhomenie-microej) [reply vee-port/extensions/front-panel/build.gradle.kts:10]: Yes indeed, done.

File: gradle/libs.versions.toml (hunk 3 above)
Comment (tdelhomenie-microej) [reply gradle/libs.versions.toml]: Done.

File: gradle/libs.versions.toml (hunk 4 above)
Comment (tdelhomenie-microej) [reply gradle/libs.versions.toml]: Done.

File: gradle/libs.versions.toml (hunk 7 above)
Comment (tdelhomenie-microej) [reply gradle/libs.versions.toml]: Done. Variable added.

File: vee-port/extensions/front-panel/src/main/resources/platform.fp (hunk 8 above)
Comment (tdelhomenie-microej) [reply vee-port/extensions/front-panel/src/main/resources/platform.fp]: Done.

**Review by gbalan-microej — APPROVED** [review]

File: gradle/libs.versions.toml [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+# APIs
+audio = { group = "ej.api", name = "audio", version = "1.0.0" }
+edc = { group = "ej.api", name = "edc", version = "1.3.7" }
+bon = { group = "ej.api", name = "bon", version = "1.4.3" }
+drawing = { group = "ej.api", name = "drawing", version = "1.0.5" }
+ecom-wifi = { group = "ej.api", name = "ecom-wifi", version = "2.1.2" }
+event = { group = "ej.api", name = "event", version = "2.1.0" }
+fs = { group = "ej.api", name = "fs", version = "2.0.6" }
+gnss = { group = "ej.api", name = "gnss", version = "2.0.0" }
+microui = { group = "ej.api", name = "microui", version = "3.5.0" }
```
Comment (gbalan-microej) [inline gradle/libs.versions.toml:15]: use latest version: 3.6.0

File: gradle/libs.versions.toml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+# Architecture
+architecture = { group = "com.microej.architecture.CM4.CM4hardfp_GCC48", name = "flopi4G25", version = "8.2.0" }
+
+# Packs
+pack-ui-architecture = { group = "com.microej.architecture.CM4.CM4hardfp_GCC48", name = "flopi4G25-ui-pack", version.ref = "pack-ui" }
+pack-ui = { group = "com.microej.pack.ui", name = "ui-pack", version.ref = "pack-ui" }
+pack-fs = { group = "com.microej.pack", name = "fs", version = "6.0.4" }
+pack-net = { group = "com.microej.pack.net", name = "net-pack", version = "11.0.2" }
+
```
Comment (gbalan-microej) [inline gradle/libs.versions.toml:30]: suggestion: add `pack-vg = { group = "com.microej.pack.vg", name = "vg-pack", version.ref = "1.7.0" }` to be compliant with the lines `vg-testsuite`

File: gradle/libs.versions.toml (hunk 10 above)
Comment (gbalan-microej) [inline gradle/libs.versions.toml:15]: suggestion: add `microvg = { group = "ej.api", name = "microvg", version = "1.5.0" }` to be compliant with the line `vg-testsuite`

File: gradle/libs.versions.toml [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+gnss-init-api = { group = "com.microej.pack.gnss", name = "gnss-init-api", version = "1.0.0" }
+gnss-testsuite = { group = "com.microej.pack.gnss", name = "gnss-testsuite", version = "2.0.0" }
+net-testsuite = { group = "com.microej.pack.net", name = "net-1_1-testsuite", version = "4.1.1" }
+net-wifi-testsuite = { group = "com.microej.pack.net", name = "net-1_1-testsuite-wifi", version = "2.1.1" }
+net-openjdk-testsuite = { group = "com.microej.pack.net", name = "net-1_1-testsuite-openjdk", version = "1.1.1" }
+security-testsuite = { group = "com.microej.pack.net", name = "security-1_7-testsuite", version = "1.7.0" }
+ssl-testsuite = { group = "com.microej.pack.net", name = "net-ssl-2_2-testsuite", version = "5.0.0" }
+ssl-openjdk-testsuite = { group = "com.microej.pack.net", name = "net-ssl-2_2-testsuite-openjdk", version = "2.0.0" }
+ui-testsuite = { group = "com.microej.pack.ui", name = "ui-testsuite", version.ref = "pack-ui" }
+vg-testsuite = { group = "com.microej.pack.vg", name = "vg-testsuite", version = "3.0.0" }
```
Comment (gbalan-microej) [inline gradle/libs.versions.toml:56]: use latest version: 4.0.0

File: gradle/libs.versions.toml [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+event = { group = "ej.api", name = "event", version = "2.1.0" }
+fs = { group = "ej.api", name = "fs", version = "2.0.6" }
+gnss = { group = "ej.api", name = "gnss", version = "2.0.0" }
+microui = { group = "ej.api", name = "microui", version = "3.5.0" }
+net = { group = "ej.api", name = "net", version = "1.1.1" }
+security = { group = "ej.api", name = "security", version = "1.7.0" }
+ssl = { group = "ej.api", name = "ssl", version = "2.2.1" }
+sni = { group = "ej.api", name = "sni", version = "1.4.0" }
+watchdog-timer = { group = "ej.api", name = "watchdog-timer", version = "2.0.0" }
+
```
Comment (gbalan-microej) [inline gradle/libs.versions.toml:21]: according to me, you can add `trace = { group = "ej.api", name = "trace", version = "1.1.1" }`

File: gradle/libs.versions.toml [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+# Packs
+pack-ui-architecture = { group = "com.microej.architecture.CM4.CM4hardfp_GCC48", name = "flopi4G25-ui-pack", version.ref = "pack-ui" }
+pack-ui = { group = "com.microej.pack.ui", name = "ui-pack", version.ref = "pack-ui" }
+pack-fs = { group = "com.microej.pack", name = "fs", version = "6.0.4" }
+pack-net = { group = "com.microej.pack.net", name = "net-pack", version = "11.0.2" }
+
+# Libraries
+frontpanel-framework = { group = "ej.tool.frontpanel", name = "framework", version = "1.1.0" }
+frontpanel-widget = { group = "ej.tool.frontpanel", name = "widget", version = "4.0.2" }
```
Comment (gbalan-microej) [inline gradle/libs.versions.toml:33]: I don't know how to do this or if it's a good solution, but the name of this component should reflect the fact that it's only compatible with the ui pack. `frontpanel-ui-widget`?

File: vee-port/build.gradle.kts [hunk 15]
```diff
@@ -1,16 +1,18 @@
...
 dependencies {
-    microejArchitecture("com.microej.architecture.CM4.CM4hardfp_GCC48:flopi4G25:8.2.0")
+    microejArchitecture(libs.architecture)
 
-    //Uncomment the microejPack dependency to set a Pack
-    //microejPack("com.mycompany:mypack:1.0.0")
+    // Comment the packs you don't need
+    microejPack(libs.pack.ui.architecture)
+    microejPack(libs.pack.fs)
+    microejPack(libs.pack.net)
```
Comment (gbalan-microej) [inline vee-port/build.gradle.kts:15]: and also vg

File: gradle/libs.versions.toml (hunk 11 above)
Comment (gbalan-microej) [reply gradle/libs.versions.toml:30]: actually there are a lot of packs like audio, watchdog... TBD

File: vee-port/validation/vg/build.gradle.kts [hunk 16]
```diff
@@ -27,13 +27,14 @@ testing {
...
-
-                implementation("ej.api:microui:3.5.0")
-                implementation("ej.api:drawing:1.0.5")
-                implementation("com.microej.pack.vg:vg-testsuite:3.0.0")
+                implementation(libs.junit)
+                implementation(libs.junit.platform)
+
+                implementation(libs.edc)
+                implementation(libs.bon)
+                implementation(libs.microui)
```
Comment (gbalan-microej) [inline vee-port/validation/vg/build.gradle.kts:35]: missing microvg

File: vee-port/configuration.properties (hunk 2 above)
Comment (gbalan-microej) [reply vee-port/configuration.properties]: nope: it is required when you installing the ui pack

Comment (tdelhomenie-microej) [reply vee-port/configuration.properties]: ok, done.

File: gradle/libs.versions.toml (hunk 10 above)
Comment (tdelhomenie-microej) [reply gradle/libs.versions.toml]: Done.

Comment (tdelhomenie-microej) [reply gradle/libs.versions.toml]: Done.

File: gradle/libs.versions.toml (hunk 12 above)
Comment (tdelhomenie-microej) [reply gradle/libs.versions.toml]: Done.

File: gradle/libs.versions.toml (hunk 14 above)
Comment (tdelhomenie-microej) [reply gradle/libs.versions.toml]: Done.

File: vee-port/build.gradle.kts (hunk 15 above)
Comment (tdelhomenie-microej) [reply vee-port/build.gradle.kts:18]: Done.

File: vee-port/validation/vg/build.gradle.kts (hunk 16 above)
Comment (tdelhomenie-microej) [reply vee-port/validation/vg/build.gradle.kts]: Done.

**Review by gbalan-microej — APPROVED** [review]

**Pull request merged**